components/board/genericlinux/data.go

```diff
@@ -14,38 +14,6 @@ import (
 
 // adapted from https://github.com/NVIDIA/jetson-gpio (MIT License)
 
-// GPIOBoardMapping represents a GPIO pin's location locally within a GPIO chip
-// and globally within sysfs.
-type GPIOBoardMapping struct {
-	GPIOChipDev    string
-	GPIO           int
-	GPIOGlobal     int
-	GPIOName       string
-	PWMSysFsDir    string // Absolute path to the directory, empty string for none
-	PWMID          int
-	HWPWMSupported bool
-}
-
-// PinDefinition describes board specific information on how a particular pin can be accessed
-// via sysfs along with information about its PWM capabilities.
-type PinDefinition struct {
-	GPIOChipRelativeIDs map[int]int    // ngpio -> relative id
-	GPIONames           map[int]string // e.g. ngpio=169=PQ.06 for claraAGXXavier
-	GPIOChipSysFSDir    string
-	PinNumberBoard      int
-	PinNumberBCM        int
-	PinNameCVM          string
-	PinNameTegraSOC     string
-	PWMChipSysFSDir     string // empty for none
-	PWMID               int    // -1 for none
-}
-
-// BoardInformation details pin definitions and device compatibility for a particular board.
-type BoardInformation struct {
-	PinDefinitions []PinDefinition
-	Compats        []string
-}
-
 // A NoBoardFoundError is returned when no compatible mapping is found for a board during GPIO board mapping.
 type NoBoardFoundError struct {
```

Comment:
I'm open to disagreement on this, but my instinct is to keep this type with the others (and to keep the `Error` and `noBoardError` functions with this struct, too)